scene/2d/sprite_2d.cpp

```diff
@@ -56,6 +59,20 @@ Point2 Sprite2D::_edit_get_pivot() const {
 bool Sprite2D::_edit_use_pivot() const {
 	return true;
 }
+
+void Sprite2D::_edit_set_rect(const Rect2 &p_rect) {
+	if (texture.is_null()) {
+		return;
+	}
+	if (!(region_enabled && Input::get_singleton()->is_key_label_pressed(Key::CTRL))) {
+		Node2D::_edit_set_rect(p_rect);
+		return;
+	}
+	Point2 pivot = p_rect.position - (centered ? _get_rect_offset(p_rect.size) : Vector2());
+	offset = pivot; // This is needed as we have to keep offset being Vector2()
```

Comment:
It makes the sprite jump if it had non-zero offset:

https://github.com/user-attachments/assets/78a4435e-388d-440f-af5d-03242536e646